llvm/lib/Target/RISCV/MCTargetDesc/RISCVMatInt.cpp

```diff
@@ -510,10 +510,12 @@ int getIntMatCost(const APInt &Val, unsigned Size, const MCSubtargetInfo &STI,
   int Cost = 0;
   for (unsigned ShiftVal = 0; ShiftVal < Size; ShiftVal += PlatRegSize) {
     APInt Chunk = Val.ashr(ShiftVal).sextOrTrunc(PlatRegSize);
+    if (FreeZeroes && Chunk == 0)
```

Comment:
It's smaller in code size to check for 0 using the int64_t returned from `Chunk.getSExtValue()` since there won't be code emitted for the non-small APInt case.